raft/log.go

```diff
@@ -138,6 +138,19 @@ func (l *raftLog) findConflict(ents []pb.Entry) uint64 {
 	return 0
 }
 
+// findConflictByTerm takes an (index, term) pair (indicating a conflicting log
+// entry on a follower during an append) and finds the last index for that term
+// (or the first index of the log, whichever is larger).
+func (l *raftLog) findConflictByTerm(index uint64, term uint64) uint64 {
+	for {
+		if logTerm, err := l.term(index); err == nil && logTerm > term {
```

Comment:
The `err == nil` case is concerning me. Do we have test cases that exercise it?

At a minimum, I think it would be valuable to create a `firstIndex := err != nil` variable to hint to readers what that error means in this context. And then structure this all as:
```go
func (l *raftLog) findLastConflictingIndexAtOrBelowTerm(index uint64, searchTerm uint64) uint64 {
    for {
            logTerm, err := l.term(index)
            firstIndex := err != nil
            if logTerm <= searchTerm || firstIndex {
                break
            }
            index--
    }
    return index
}
```

Reply:
Done.